consensus/bor/heimdall/span/spanner.go

```diff
@@ -3,82 +3,57 @@ package span
 import (
 	"context"
 	"encoding/hex"
-	"math"
 	"math/big"
 
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/common/hexutil"
-	"github.com/ethereum/go-ethereum/consensus/bor/abi"
-	"github.com/ethereum/go-ethereum/consensus/bor/api"
-	"github.com/ethereum/go-ethereum/consensus/bor/statefull"
 	"github.com/ethereum/go-ethereum/consensus/bor/valset"
 	"github.com/ethereum/go-ethereum/core"
 	"github.com/ethereum/go-ethereum/core/state"
 	"github.com/ethereum/go-ethereum/core/types"
-	"github.com/ethereum/go-ethereum/internal/ethapi"
+	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/params"
 	"github.com/ethereum/go-ethereum/rlp"
 	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/go-ethereum/tests/bor/contracts"
 )
 
 type ChainSpanner struct {
-	ethAPI                   api.Caller
-	validatorSet             abi.ABI
+	validatorSet             *contracts.BorValidatorSet
 	chainConfig              *params.ChainConfig
 	validatorContractAddress common.Address
 }
 
-func NewChainSpanner(ethAPI api.Caller, validatorSet abi.ABI, chainConfig *params.ChainConfig, validatorContractAddress common.Address) *ChainSpanner {
```

Comment:
could we keep interfaces like `ethAPI api.Caller` or introduce new ones? I wouldn't expect to make an http connection inside such constructors.

Reply:
like `backend bind.ContractBackend`